### ISSUE TITLE
Parsing of numbers from json loses precision

### ISSUE THREAD

**Issue by goncalopp** [s4w3d0ff/python-poloniex#31]
For example, for a orderbook:

    {"asks":[["0.00781195",0.12377319] ...

`0.12377319` will be converted to a float, which is a lossy process. IMHO, this should be handled with Decimal, to avoid mangling the number
There's a nice overview of the matter [on this blog](http://www.daveoncode.com/2014/10/21/python-reading-numbers-from-json-without-loss-of-precision-using-decimal-class-for-data-processing/)

Basically, use `json.loads(..., parse_float=Decimal)`

If you agree with this, I can send a PR

**Comment by s4w3d0ff** [comment]
Thank you, you are 100% correct, I had noticed this behavior a few months back and opened a support ticket with polo to try and figure out where the rounding was coming from (thinking it might have to do with something on their end), I had never really figured it out and the bug 'disappeared.' Seems this was the initial problem to begin with.

TBH I had no idea the json module could convert values while decoding.

I'll make the commit directly to master and publish a release sometime today.

**Comment by s4w3d0ff** [comment]
Poloniex data sets are **not** consistent across api commands, some return numbers others return strings when you would expect numbers.
```
{u'bids': [[u'0.00062074', Decimal('2.41647066')],,,],}
```
This PR shouldn't hurt anything, but if poloniex decides to switch its unicode numbers to floats/decimals, things might get a little buggy/inconsistent if users are doing high precision math from these raw data sets.

**Comment by goncalopp** [comment]
>thinking it might have to do with something on their end
Actually, you're right here. As you've noticed, they're encoding number-likes as both json strings and number *in the same message* - exchange rate is a string, amount is a float. 

I'm not sure whether that reflects that the original amount is stored as a float, or they botched the REST code - in either case, it seems some shoddy codding on their end.

With that being said, we (library developers) should strive to give as much precision as we possibly can with what the service gives us.

>if poloniex decides to switch its unicode numbers to floats/decimals, things might get a little buggy/inconsistent
That's a good point. Maybe we can `parse_float=str`, and let the user deal with it?

**Comment by s4w3d0ff** [comment]
I agree, parse them as strings and let the user choose how they want to convert is probably best. Not sure if just `str `will do it for both python 2 and 3 (maybe unicode or a futures class) but I will push the commit to #32 and test asap

**Comment by goncalopp** [comment]
`str` should work as intended in python3. In fact, `unicode` doesn't exist anymore

**Comment by s4w3d0ff** [comment]
ah, sounds good

**Comment by s4w3d0ff** [comment]
Here is my python 2 test
```
Python 2.7.12 (default, Jul  1 2016, 15:12:24) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import poloniex
>>> polo = poloniex.Poloniex()
>>> polo.returnOrderBook('BTC_CGA')
{u'bids': [[u'0.00062355', '2.40558095'],,,
```

**Comment by s4w3d0ff** [comment]
OK... so #33 should fix the string inconsistency between python 2 and 3:

```
Python 3.5.2 (default, Sep 10 2016, 08:21:44) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from poloniex import Poloniex
>>> Poloniex().returnOrderBook('BTC_CGA')
{'bids': [['0.00062461', '4.00256164'],
```

```
Python 2.7.12 (default, Jul  1 2016, 15:12:24) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from poloniex import Poloniex
>>> Poloniex().returnOrderBook('BTC_CGA')
{u'bids': [[u'0.00062461', u'4.00256164'],
```

I think we are good for a release.

**Comment by goncalopp** [comment]
I noticed you released 0.2.1
Thank you for your efforts, it's great to be able to have proper decimal precision!
In case it interests you, I'm pulling this into the [experimental branch of mexbtcapi](https://github.com/goncalopp/mexbtcapi/tree/experimental)

**Comment by s4w3d0ff** [comment]
No problem, thanks for the help!

**Comment by s4w3d0ff** [comment]
I am starting not to like what we did :roll_eyes: I'm probably going to make this all optional and allow the user to define what datatype they want to use for numbers. Probably should have been done in the first place... :P

**Comment by s4w3d0ff** [comment]
see #68
